tests/codemods/test_with_threading_lock.py

```diff
@@ -86,3 +86,108 @@ def test_no_effect_multiple_with_clauses(self, tmpdir, klass):
     ...
 """
         self.run_and_assert(tmpdir, input_code, input_code)
+
+
+class TestThreadingNameResolution(BaseSemgrepCodemodTest):
+    codemod = WithThreadingLock
+
+    @pytest.mark.parametrize(
+        "input_code,expected_code",
+        [
+            (
+                """from threading import Lock
+lock = 1
+with Lock():
+    ...
+""",
+                """from threading import Lock
+lock = 1
+lock_1 = Lock()
+with lock_1:
+    ...
+""",
+            ),
+            (
+                """from threading import Lock
+from something import lock
+with Lock():
+    ...
+""",
+                """from threading import Lock
+from something import lock
+lock_1 = Lock()
+with lock_1:
+    ...
+""",
+            ),
+            (
+                """import threading
+lock = 1
+def f(l):
+    with threading.Lock():
+        return [lock_ for lock_ in l]
```

Comment:
Can you change `lock_` to `lock_1` to show this won't shadow it? That was the original purpose of this example.

Reply:
ah good catch, will update